### PR TITLE
feat(log_store): use new `Consumer`

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Switch to new Consumer

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/log-store/src/kafka/log_store.rs [hunk 1]
```diff
@@ -228,35 +229,27 @@ impl LogStore for KafkaLogStore {
...
-            .with_max_batch_size(self.max_batch_bytes as i32)
-            .with_max_wait_ms(self.consumer_wait_timeout.as_millis() as i32)
-            .build();
+        let index = RegionWalRange::new(range, self.max_batch_bytes);
 
-        debug!(
-            "Built a stream consumer for ns {} to consume entries in range [{}, {}]",
-            provider, start_offset, end_offset
-        );
+        // Safety: Must be ok.
```
Comment (fengjiachun) [inline src/log-store/src/kafka/log_store.rs:245]: Really?
```suggestion
        // Safety: (weny)Must be ok.
```

File: src/log-store/src/kafka/consumer.rs [hunk 2]
```diff
@@ -60,40 +61,61 @@ struct FetchResult {
     used_offset: i64,
 }
 
+const MAX_BATCH_SIZE: usize = 52428800;
+const AVG_RECORD_SIZE: usize = 256 * 1024;
```
Comment (fengjiachun) [inline src/log-store/src/kafka/consumer.rs:65]: Can they be `ReadableSize`?

File: src/log-store/src/kafka/log_store.rs (hunk 1 above)
Comment (WenyXu) [reply src/log-store/src/kafka/log_store.rs:245]: Yes, all required fields are set.

File: src/log-store/src/kafka/consumer.rs (hunk 2 above)
Comment (WenyXu) [reply src/log-store/src/kafka/consumer.rs]: Resolved in [35280f7](https://github.com/GreptimeTeam/greptimedb/pull/4510/commits/35280f773a591ea6d8a74d570c8722d02c930f6e)

**Review by fengjiachun — APPROVED** [review]
LGTM

**Comment by WenyXu** [comment]
cherry pick the commit into #4565

**Pull request closed without merging**